src/store/ChartStore.js

```diff
@@ -219,6 +219,7 @@ class ChartStore {
         // TODO: excluded studies
 
         this.loader.show();
+        this.loader.setState('chart-engine');
```

Comment:
you are placing the status messages (all of them) in the wrong place. 

Where you are placing the chart engine loading message - the chart engine has already been loaded.